File: Homework/quoc_khanh/217.ContainsDuplicate.py
#https://leetcode.com/problems/contains-duplicate

from typing import List

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        distintNums = set([])
        for num in nums:
            if num in distintNums:
                return True
            else:
                distintNums.add(num)
        return False

s = Solution()
print(s.containsDuplicate([1]) == False)
print(s.containsDuplicate([1, 2]) == False)
print(s.containsDuplicate([1, 1]) == True)
print(s.containsDuplicate([1,2,3,1]) == True)
print(s.containsDuplicate([1,2,3,4]) == False)
print(s.containsDuplicate([1,1,1,3,3,4,3,2,4,2]) == True)

File: Homework/Hung_Tran/Warmup/217_contains_duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function (nums) {
  for (let i = 0; i < nums.length; i++) {
    for (let j = 0; j < nums.length; j++) {
      if (j != i) {
        if (nums[j] == nums[i]) {
          return true;
        }
      }
    }
  }
  return false;
};
